### PR TITLE
Add IoT community

### PR DESCRIPTION
**Pull request by kkartunov**
@birdofpreyru @ajefts This is the result code base of all IoT challenges.It is up-to-date with `new-develop`, tests, build etc functional.
@ajefts We could merge now so it drifts slowly to master/production or we can wait for 'Contentful' integration. Please, advice?

### CODE REVIEW

File: src/shared/actions/page/communities/iot/assets.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+function toggleGrid() {
+  return 'grid';
+}
+
+/**
+ * toggle list display mode
+ * */
+function toggleList() {
+  return 'list';
+}
```
Comment (birdofpreyru) [inline src/shared/actions/page/communities/iot/assets.js]: Bad design. Should be:
- 1 variable in the Redux store keeps the current view mode (`grid`, `list`, etc.)
- A map exported from this actions module gives all valid values for that variable.
- A single action `toggleView(mode)` switches page into the desired view.

File: src/shared/components/tc-communities/communities/iot/AssetDetail/ShareSocial.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import EmailIcon from '../../../../../../assets/images/social/icon_email.svg';
+import MoreIcon from '../../../../../../assets/images/social/icon_plus.svg';
+
+import './social_media.scss';
+
+export default class ShareSocial extends React.Component {
+  componentDidMount() {
+    /* TODO: This is some tooltip solution added in the related challenge,
+     * it should be replaced by the standard tooltip component employed into
+     * the repo. */
```
Comment (birdofpreyru) [inline src/shared/components/tc-communities/communities/iot/AssetDetail/ShareSocial.jsx]: +1

File: src/shared/components/tc-communities/communities/iot/AssetDetail/ShareSocial.jsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import React from 'react';
+
+import TwitterIcon from '../../../../../../assets/images/social/icon_twitter.svg';
+import PrintIcon from '../../../../../../assets/images/social/icon_print.svg';
+import EmailIcon from '../../../../../../assets/images/social/icon_email.svg';
+import MoreIcon from '../../../../../../assets/images/social/icon_plus.svg';
+
+import './social_media.scss';
+
+export default class ShareSocial extends React.Component {
```
Comment (birdofpreyru) [inline src/shared/components/tc-communities/communities/iot/AssetDetail/ShareSocial.jsx]: We also have social share buttons in the sidebar of challenge details page. We should look to unify / reuse the social share block in future.

File: src/shared/components/tc-communities/communities/iot/JoinSection/index.jsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+const JoinSection = ({ baseUrl }) => (
+  <Section
+    title="Are you ready for Predix?"
+    theme={{
+      container: styles.sectionContainer,
+    }}
+  >
+    <div styleName="buttonContainer">
+      <JoinCommunity competeUrl={`${baseUrl}/challenges?communityId=iot`} label="Join the Topcoder Community for Predix" />
```
Comment (birdofpreyru) [inline src/shared/components/tc-communities/communities/iot/JoinSection/index.jsx]: `competeUrl`? I believe, `JoinCommunity` button does not have such prop.
It only takes theme, label, and communityId, which should be joined. Do I miss something?

Comment (kkartunov) [reply src/shared/components/tc-communities/communities/iot/JoinSection/index.jsx]: `competeUrl` is some artifact from the last F2F as there was requirement to show other button linked /challenges for registered community users but this was not implemented good and I removed it.

**Comment by kkartunov** [comment]
I am waiting on the decision from @ajefts then we can merge and fix those minor issues latter...

**Comment by birdofpreyru** [comment]
@kkartunov I believe, @ajefts is fine to merge it in as soon as possible, provided it does no break anything outside of the community; and then make further enhancements / refactoring as necessary.

**Comment by kkartunov** [comment]
Me too @birdofpreyru . He is probably busy with something else. Code does not break anything outside of the community so we can proceed.

**Comment by ajefts** [comment]
please go ahead

**Pull request merged**